elastic-buffer-sim/src/Bittide/Plot.hs

```diff
@@ -368,29 +396,39 @@ simulateTopology ccc margin framesize graph settings = do
   (0, n) = bounds $ unboundGraph $ graph
 
 -- | Plots the datacount of an elastic buffer and marks those parts of
--- the plots that are reported to be stable by the stability checker
--- for the repsective buffer.
+-- the plots that are reported to be stable/centered by the stability
+-- checker for the repsective buffer.
 plotEbData ::
-  (ToJSON t, ToJSON d) => [(t, (d, Bool))] -> Matplotlib
+  (ToJSON t, ToJSON d) => [(t, (d, (Bool, Bool)))] -> Matplotlib
 plotEbData xs = foldPlots markedIntervals % ebPlot
  where
-  mark = (@@ [ o1 "g-", o2 "linewidth" (10 :: Int)])
+  markGreen = (@@ [ o1 "g-", o2 "linewidth" (10 :: Int)])
+  markBlue = (@@ [ o1 "b-", o2 "linewidth" (10 :: Int)])
   ebPlot = uncurry plot $ unzip (pData <$> xs)
 
   markedIntervals =
-    mark . uncurry plot . unzip
-      <$> stableIvs [] False xs
+    (\(mark, ys) -> mark $ uncurry plot $ unzip ys)
+      <$> stableIvs [] False False xs
 
-  stableIvs a _       []     = a
-  stableIvs a collect (x:xr) = stableIvs a' (stable x) xr
+  stableIvs a _  _  []     = a
+  stableIvs a as ac (x:xr) = stableIvs a' (stable x) (centered x) xr
    where
-    a' | stable x       && not collect = [pData x] : a
-       | stable x       && collect     = (pData x : y) : yr
-       | not (stable x) && collect     = reverse y : yr
-       | otherwise                     = a
-    y : yr = a
-
-  stable (_, (_, s)) = s
+    a' |      stable x  && not (centered x) && not as           =
+         (markBlue,  [pData x]  ) : a
+       |      stable x  &&      centered x  &&           not ac =
+         (markGreen, [pData x]  ) : a
+       |      stable x  && not (centered x) &&     as &&     ac =
+         (markBlue,  [pData x]  ) : (m, reverse y) : yr
+       |      stable x                      &&     as           =
+         (m,         pData x : y) : yr
+       | not (stable x)                     &&     as           =
+         (m,         reverse y  ) : yr
+       | otherwise                                              =
```

Comment:
It's probably sensible to break the 80/90 limit here. It's not a complicated piece of code, but it's also hard to format it properly. (One time only :eyes:.)

```suggestion
    a' |      stable x  && not (centered x) && not as           = (markBlue,  [pData x]  ) : a
       |      stable x  &&      centered x  &&           not ac = (markGreen, [pData x]  ) : a
       |      stable x  && not (centered x) &&     as &&     ac = (markBlue,  [pData x]  ) : (m, reverse y) : yr
       |      stable x                      &&     as           = (m,         pData x : y) : yr
       | not (stable x)                     &&     as           = (m,         reverse y  ) : yr
       | otherwise                                              = a
```

Reply:
Fixed.